### PR TITLE
feature/SM-6050: add new fields to permit message

### PR DESCRIPTION
**Pull request by ccurran01**
## Description

add new fields to the permit message received by Open Data subscribers

Data: https://github.com/departmentfortransport/street-manager-data/pull/114
Event Subscriber: https://github.com/departmentfortransport/street-manager-event-subscriber/pull/14
Event Notifier: https://github.com/departmentfortransport/street-manager-event-notifier/pull/30

## Coding Standards

Style guide: https://github.com/departmentfortransport/street-manager/wiki/Street-Manager-Style-Guide
Coding best practices: https://github.com/departmentfortransport/street-manager/wiki/Street-Manager-Coding-Best-Practices

## Checklist

- [ ] All unit tests passing
- [ ] Code coverage suitable
- [ ] Branch named {feature|hotfix|task}/{SM-.*}
- [ ] If your pull request depends on any other, please link them
- [ ] Changes approved by your team
- [ ] Changes approved by another team
- [ ] Significant events are being logged e.g. error scenarios
- [ ] Commit messages are meaningful
- [ ] Target is green https://circleci.com/gh/departmentfortransport/street-manager-event-notifier/tree/master
- [ ] Int UI tests passing https://circleci.com/gh/departmentfortransport/street-manager-int/tree/master
- [ ] Add `DO NOT MERGE` if you want to postpone merge

### CODE REVIEW

File: src/daos/permitConditionDao.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import * as Knex from 'knex'
+import { injectable } from 'inversify'
+import { PermitPermitCondition } from 'street-manager-data'
+
+@injectable()
+export default class PermitConditionDao {
```
Comment (caoimheb) [inline src/daos/permitConditionDao.ts:6]:
```suggestion
export default class PermitPermitConditionDao {
```

File: tests/fixtures/permitConditionFixtures.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { PermitPermitCondition, RefPermitConditionType } from 'street-manager-data'
+
+export function generatePermitCondition(permitVersionId: number, permitConditionId: RefPermitConditionType = RefPermitConditionType.NCT01a): PermitPermitCondition {
```
Comment (caoimheb) [inline tests/fixtures/permitConditionFixtures.ts:3]:
```suggestion
export function generatePermitPermitCondition(permitVersionId: number, permitConditionId: RefPermitConditionType = RefPermitConditionType.NCT01a): PermitPermitCondition {
```

**Comment by caoimheb** [comment]
Looks good 👍  Just a couple of comments on naming

**Review by caoimheb — APPROVED** [review]

File: src/filters.ts [hunk 3]
```diff
@@ -96,6 +96,14 @@ export function locationTypeFilter(locationTypeId: number): string {
   return LocationType[locationTypeId]
 }
 
+export function permitStatusFilter(permitStatusId: number): string {
+  return RefPermitStatus[permitStatusId]
```
Comment (cpaul92) [inline src/filters.ts:100]: Shouldn't we specify the text we want to display like we do for `WorksStatusType` here?

Comment (ccurran01) [reply src/filters.ts:100]: So these type of fields have a few different approaches going on at the moment. 

Work status is using a frontend display type value. 
Some fields like traffic_management_type have a frontend display value but also have a _ref which just uses the reference string.
And others like this new permitStatus filter are just using the reference string, so values like granted or modification_requested. 

Was a bit confused about what the right thing to do here was but Andy has said that for furture fields we're going to default to ref type only with no _ref field. 

That code will give us the behaviour that we want though we could RefPermitStatus again to another enum which would, at least for now, be a redefinition of the RefPermitStatus?

**Review by cpaul92 — APPROVED** [review]

**Pull request merged**